src/odemis/gui/comp/overlay/world.py

```diff
@@ -2662,3 +2663,253 @@ def draw(self, ctx, shift=(0, 0), scale=1.0):
+                    # Clicked inside selection or near line, so start dragging
+                    self.start_drag()
+                else:
+                    # Clicked on an edit point (e.g. an edge or start or end point), so edit
+                    self.start_edit(hover)
+
+            self._view_to_phys()
+            self.cnvs.update_drawing()
+        else:
+            WorldOverlay.on_left_down(self, evt)  # skip event
+
+    def on_left_up(self, evt):
+        """
+        Check if left click was in ROA. If so, activate the overlay. Otherwise, deactivate.
+        """
+        abort_roa_creation = (self._coordinates.value == UNDEFINED_ROI and
+                              max(self.get_height(), self.get_width()) < gui.SELECTION_MINIMUM)
+        if abort_roa_creation:
+            # Process aborted by clicking in the viewport
+            # VA did not change, so notify explicitly to make sure aborting the process works
+            self._coordinates.notify(UNDEFINED_ROI)
+        else:
+            # Activate/deactivate region
+            self._view_to_phys()
+            # .get_physical_sel() normalizes the rectangles, i.e. swaps top and bottom coordinates.
+            # That doesn't work here, the top coordinate needs to be larger than the bottom coordinate,
+            # so use .p_start_pos and .p_end_pos instead.
+            rect = self.p_start_pos + self.p_end_pos
+            pos = self.cnvs.view_to_phys(evt.Position, self.cnvs.get_half_buffer_size())
+            self.active.value = True if self._is_point_in_rect(pos, rect) else False
+
+            # Update ._coordinates VA
+            if self.active.value:
+                self._coordinates.value = rect
+
+        # SelectionMixin._on_left_up has some functionality which does not work here, so only call the parts
+        # that we need
+        self.clear_drag()
+        self.selection_mode = SEL_MODE_NONE
+        self.edit_hover = None
+
+        self.cnvs.update_drawing()  # Line width changes in .draw when .active is changed
+        self.cnvs.reset_default_cursor()
+        WorldOverlay.on_left_up(self, evt)
+
+    def _is_point_in_rect(self, p, rect):
+        """
+        Checks if point is contained in rectangle.
+        p (tuple of 2 floats): x, y coordinates of point in m
+        rect (tuple of 4 floats): l, t, r, b positions of rectangle in m
+        """
+        l, t, r, b = rect
+        if l <= p[0] <= r and b <= p[1] <= t:
+            return True
+        else:
+            return False
+
+    def draw(self, ctx, shift=(0, 0), scale=1.0, line_width=None, dash=True):
+        """ Draw the selection as a rectangle. Exactly the same as parent function except that
+         it has an adaptive line width (wider if the overlay is active). """
+        line_width = 5 if self.active.value else 2
+        super(FastEMROAOverlay, self).draw(ctx, shift, scale, line_width, dash)
+
+
+class FastEMROCOverlay(WorldSelectOverlay):
```

Comment:
I'm not sure about that. There are sufficient differences between the classes. I could have this interface

```
FastEMROIOverlay(cnvs, coordinates, colour=gui.SELECTION_COLOUR, allow_edit=True, dash=True, label=None, activation_margin=0)
"""
allow_edit: don't allow changing region boundaries, don't change cursor when hovering on boundaries
dash: True if overlay box should be dashed
label: label of overlay box
activation_margin: margin outside box that triggers activation (for roc to allow selection from far zoom level)
"""
```

For me, it feels clearer to have two classes than all these parameters. Maybe in the future the classes will diverge more and we'll have to add more arguments.

Inheritance is an option, but doesn't change much. The common functions are init and the ._coordinates VA callback. Possibly also `_on_left_up`, the ROA has some special code here for aborting the process, but it would also not hurt the ROC (although in this case it might be confusing).

What's your preferred choice?

Reply:
Indeed, I think it's enough differences that it'd feel a little odd in a single class. I'd suggest to go with inheritance.

Reply:
Ok. I created a superclass `FastEMSelectOverlay`, but to be honest, I don't think it's making the code any easier. Maybe you have some ideas on how to improve the inheritance.